### PR TITLE
airbyte-ci: augment the report for java connectors

### PR DESCRIPTION
**Pull request by stephane-airbyte**
Today we're missing the logs (both JVM and container logs) in java connector reports.
This is creating a link to test artifacts. In the CI, the link will point to a zip file, while on a local run, it will point to a directory.

In addition, we recently added the junit XML inlined with the test standard output and error, but that didn't really work as well as we'd hoped: The reports were slow to load, they were not ordered by time, the corresponding logs were lacking. There's still a possibility they'll be useful, so rather than removing them altogether, they will be bundled in the log zip (or directory).

I'm also adding a button to copy the standard output or the standard error from a step into the clipboard.
Finally, I'm reducing the max vertical size of an expanded step, so it doesn't go over 70%, which seems much cleaner to me.

Here's an example of the result (from the child PR): https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/stephane_02-09-add_background_thread_to_track_mssql_container_status/1708056420/d4683bfb7f90675c6b9e7c6d4bbad3f98c7a7550/source-mssql/3.7.0/output.html

### CODE REVIEW

**Comment by stephane-airbyte** [comment]
* `master`
  * **#35317** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35317?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @stephane-airbyte and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/reports.py
Comment (stephane-airbyte) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/reports.py:14]: pathlib and anyio didn't play nice with each other. Since we were awaiting all the Path function, we might as well use the synchronous version from pathlib

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/java_connectors.py
Comment (postamar) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/java_connectors.py:81]: is there any point in maintaining both `with_logs` and `with_test_report`? seems like it should be one or the other

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/steps/gradle.py
Comment (postamar) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/steps/gradle.py:114]: can you put this call on a new line with a comment explaining the purpose of the env var please?

**Review by postamar — APPROVED** [review]
No major objections but I'd have @alafanechere do a review before merging

**Review by alafanechere — CHANGES_REQUESTED** [review]
What a great feature addition! And thanks for the example!

I'm requesting changes to suggest taking a slightly more generic approach.
Could we:
* Pass (test) artifact around in an `output_artifacts` attribute of `StepResults` (and renaming `output_artifact` to `output_artifacts`
* Create an for `Artifact` class with attributes like: `name` (str), `content` (a dagger `File` /  `Directory` / `Container`) , `to_save` (`bool`), `saved` (`bool`)
* Steps would instantiate `Artifact`
* The pipeline functions would collect artifacts and set them on the `Context` object
* The upload / write to disk would happen on `Context` teardown in [ `__aexit__`](https://github.com/airbytehq/airbyte/blob/main/airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/context.py#L237) according to the execution mode (local/CI).  This is where the `artifact.saved` attribute would be set to `True`
* Reports would retrieve URLs for artifacts, which are step results attributes

With this approach:
* `Step` would not mutate the filesystem with `exports`. 
* Artifacts are written / upload at a single place
* Anything can get uploaded outside of reports
* We might be able to upload containers to GCS which would be useful for debugging, 

Does this sound like a reasonable change to make?

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/reports.py
Comment (stephane-airbyte) [reply airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/reports.py:11]: unfortunately I believe it is. Also, every single use of anyio.path was doing an await, so I really didn't see anything taking advantage of its non-blocking aspect

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/java_connectors.py
Comment (stephane-airbyte) [reply airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/java_connectors.py:42]: no, only tasks that do execute the java will have test artifacts.

**Comment by stephane-airbyte** [comment]
I think it's really a lot to ask for this PR. We can file an issue to get that genericized later, but I've honestly already spent way too much time on this.
I have no idea what aexit is, and I'm not interested in knowing more about it right at this moment. This PR is mostly here because we have test issues that are blocking us and the lack of logs is contributing to the blocking aspect of this.

**Comment by postamar** [comment]
@alafanechere this sounds like a lot to ask. Shall we table this unified approach until the python connectors also generate test artifacts? At that point we'll be in a better place to refactor the `Step` model, right now it feels a bit premature. Who knows if we won't need to add something else then?

**Review by alafanechere — COMMENTED** [review]
Yes I understand it looks like a significant change.
I'm concerned about the fact that python connectors will have to rewrite a similar logic to get this feature.
Would you be 🆗 to:
* Pass artifacts via dagger `File` / `List[File]` / `Directory` (whatever fits your need) in your `test_output_artifact`?
* Make the Report write the `test_output_artifact` to disk before upload?

I'd ideally prefer the Step not to write to the file system state during a pipeline as could can have some side effects.

**Comment by alafanechere** [comment]
@stephane-airbyte I implemented what I suggested here https://github.com/airbytehq/airbyte/pull/35479

**Review by alafanechere — APPROVED** [review]
Thanks for merging everything together. LGTM as long as the unit test pass.

Can we manually check the following before merging:
* Manually run a test workflow from this branch
* Manually run a pre-release to make sure we did not break the publish pipeline.

I'll take care of it

**Comment by alafanechere** [comment]
[Pre-release postgres from this branch](https://github.com/airbytehq/airbyte/actions/runs/8007246460)

**Comment by alafanechere** [comment]
[Testing `source-postgres` on this branch](https://github.com/airbytehq/airbyte/actions/runs/8007258391/job/21870930737)

**Pull request merged**
